Realm/RLMMigration.mm

```diff
@@ -142,6 +145,9 @@ - (void)execute:(RLMMigrationBlock)block {
 
         // verify uniqueness for any new unique columns before committing
         [self verifyPrimaryKeyUniqueness];
+
+        // reset schema to saved schema since it has been altered
+        RLMRealmSetSchema(_realm, savedSchema, true);
```

Comment:
Was the lack of this an existing bug on master? It'd be nice to have a test that'd break on `isPrimary` not being reset properly if there were actually any effects other than the user-visible object schema being wrong.